File: problemas/matematica/1585/1585.swift
import Foundation

let n = Int(readLine(strippingNewline: true)!)!
for _ in 0..<n {
  let entrada = readLine(strippingNewline: true)!.components(separatedBy: " ")
  let x = Int(entrada[0])!
  let y = Int(entrada[1])!

  print(String(format: "%d cm2", (x * y) / 2))
}

File: problemas/matematica/1585/1585.exs
defmodule Modulo do
  def leitura(0) do
    []
  end

  def leitura(n) do
    entrada = IO.gets("") |> String.trim |> String.split(" ")
    resposta = leitura(n - 1)

    x = entrada |> Enum.at(0) |> String.to_integer
    y = entrada |> Enum.at(1) |> String.to_integer

    [div(x * y, 2) | resposta]
  end
end

n = IO.gets("") |> String.trim |> String.to_integer
resposta = Modulo.leitura(n)
Enum.each(resposta, fn r -> IO.puts :io_lib.format("~.w cm2", [r]) end)